utils/profile-services.ts

```diff
@@ -0,0 +1,217 @@
+  ResponseErrorNotFound
+} from "italia-ts-commons/lib/responses";
+import * as e from "fp-ts/lib/Either";
+import * as te from "fp-ts/lib/TaskEither";
+import * as t from "fp-ts/lib/Task";
+import { ServicesPreferencesModeEnum } from "@pagopa/io-functions-commons/dist/generated/definitions/ServicesPreferencesMode";
+import * as a from "fp-ts/lib/Array";
+import { IResponseErrorValidation } from "@pagopa/ts-commons/lib/responses";
+import { BlockedInboxOrChannelEnum } from "@pagopa/io-functions-commons/dist/generated/definitions/BlockedInboxOrChannel";
+import { TaskEither } from "fp-ts/lib/TaskEither";
+
+const NOT_FOUND_TITLE = "ServicesPreferences not found";
+
+interface IServicePreferenceHandler {
+  readonly isMyReposability: (profile: RetrievedProfile) => boolean;
+  readonly handleProfile: (
+    profile: RetrievedProfile,
+    servicesPreferencesModel: ServicesPreferencesModel,
+    serviceId: ServiceId
+  ) => t.Task<LimitedProfile>;
+}
+
+const missingServicePreference = ResponseErrorNotFound(
+  NOT_FOUND_TITLE,
+  "The ServicesPreferences you requested was not found in the system."
+);
+
+/**
+ * Converts the RetrievedProfile model to LimitedProfile type.
+ */
+export const retrievedProfileToLimitedProfile = (
+  retrivedProfile: RetrievedProfile,
+  senderAllowed: boolean
+): LimitedProfile => ({
+  preferred_languages: retrivedProfile.preferredLanguages,
+  sender_allowed: senderAllowed
+});
+
+/**
+ * Whether the sender service is allowed to send
+ * messages to the user identified by this profile
+ */
+export const isSenderAllowed = (
+  blockedInboxOrChannels:
+    | RetrievedProfile["blockedInboxOrChannels"]
+    | undefined,
+  serviceId: ServiceId
+): boolean =>
+  blockedInboxOrChannels === undefined ||
+  blockedInboxOrChannels[serviceId] === undefined ||
+  blockedInboxOrChannels[serviceId].indexOf(BlockedInboxOrChannelEnum.INBOX) <
+    0;
+
+const findServicePreference = (
+  servicesPreferencesModel: ServicesPreferencesModel,
+  profile: RetrievedProfile,
+  serviceId: ServiceId
+): te.TaskEither<IResponseErrorNotFound, RetrievedServicePreference> =>
+  te
+    .tryCatch(
+      () =>
+        asyncIterableToArray(
+          servicesPreferencesModel.getQueryIterator({
+            parameters: [
+              {
+                name: "@fiscalCode",
+                value: profile.fiscalCode
+              },
+              {
+                name: "@settingsVersion",
+                value: profile.servicePreferencesSettings.version
+              },
+              {
+                name: "@serviceId",
+                value: serviceId
+              }
+            ],
+            query: `
+                SELECT *
+                FROM c 
+                WHERE 
+                  c.fiscalCode = @fiscalCode 
+                  AND 
+                  c.settingsVersion = @settingsVersion
+                  AND
+                  c.serviceId = @serviceId
+              `
+          })
```

Comment:
This is not needed. You can simply use `servicesPreferencesModel.find` method and passing it  `documentId` and  `fiscalCode`.
You can see an implementation [here](https://github.com/pagopa/io-functions-app/blob/47808ad85bde3f7c328f12cbc925b80f06a68985/GetServicePreferences/handler.ts#L105)
Note that this query could fail  with an `IResponseErrorQuery`

Reply:
done.